### PR TITLE
Feature/GAL-600

### PR DESCRIPTION
**Pull request by Abdul-NHS**
https://nhsd-jira.digital.nhs.uk/browse/GAL-600

Implemented error handling for when no LSOAs are selected

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: src/app/views/clinic_information/ClinicInformation.js [hunk 1]
```diff
@@ -1,157 +1,210 @@
...
-    this.onChangeSelectedClinicHandler = this.onChangeSelectedClinicHandler.bind(this);
+      displayUserErrorTargetPercentage: false,
+      displayViewAllPrevInvitations: false,
+      lsoaInRange: [""],
+      rangeSelection: 1,
+      selectedLsoa: [],
+      targetFillToInputValue: 0,
+      appsToFill: 0,
+      checkAll: true,
+      rangeSelection: 1,
```
Comment (zainm97) [inline src/app/views/clinic_information/ClinicInformation.js:28]: there's two rangeSelection variables

File: src/app/views/clinic_information/ClinicInformation.js [hunk 2]
```diff
@@ -1,157 +1,210 @@
...
   onClickGoBackLinkHandler() {
-    this.context.setState({ "navigateToClinic": false })
+    this.context.setState({ navigateToClinic: false });
     // Scroll to the top of the page every time it renders the page
-    // window.scrollTo(0, 0);
+    window.scrollTo(0, 0);
+  }
+
+  onKeyUp(e) {
+    console.log("enter is pressed", e);
```
Comment (zainm97) [inline src/app/views/clinic_information/ClinicInformation.js:88]: probs need to remove log

File: src/app/views/clinic_information/ErrorBannerDetail.jsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    >
+      <h2
+        class="nhsuk-error-summary__title nhsuk-u-margin-bottom-2"
+        id="error-target-title"
+      >
+        There is a problem
+      </h2>
+      <div class="nhsuk-error-summary__body">
+        <a
+          href={`${hrefErrorMessage}`}
```
Comment (zainm97) [inline src/app/views/clinic_information/ErrorBannerDetail.jsx:27]: can this just be href={hrefErrorMessage}

File: src/app/views/clinic_information/clinic_invitation_criteria/ClinicInvitationCriteria.jsx [hunk 4]
```diff
@@ -50,7 +58,7 @@ export default function ClinicInvitationCriteria(props) {
           <div
             id="input-suffix-percentage"
             class={
-              displayUserErrorTargetPercentage
+              (displayUserErrorTargetPercentage && !lsoaTableError)
```
Comment (zainm97) [inline src/app/views/clinic_information/clinic_invitation_criteria/ClinicInvitationCriteria.jsx:61]: theres 4 of the same checks in this page. Might be worth doing a refactor to store
const meaningfulName = displayUserErrorTargetPercentage && !lsoaTableError and call that?

File: src/app/views/clinic_information/ClinicInformation.js (hunk 2 above)
Comment (Abdul-NHS) [reply src/app/views/clinic_information/ClinicInformation.js:88]: resolved in 602

**Review by zainm97 — DISMISSED** [review]
comments are nice to have but good work

File: src/app/views/clinic_information/ErrorBannerDetail.jsx (hunk 3 above)
Comment (Abdul-NHS) [reply src/app/views/clinic_information/ErrorBannerDetail.jsx:27]: this is how the id is read for a href tag

File: src/app/views/clinic_information/clinic_invitation_criteria/ClinicInvitationCriteria.jsx (hunk 4 above)
Comment (Abdul-NHS) [reply src/app/views/clinic_information/clinic_invitation_criteria/ClinicInvitationCriteria.jsx:61]: resolved in GAL-602

**Review by Amar-NHS — DISMISSED** [review]
looks good

**Review by chenglawson1 — DISMISSED** [review]

**Review by NeerjaGoyalNHS — DISMISSED** [review]

**Review by Amar-NHS — DISMISSED** [review]
Approved

**Review by Amar-NHS — APPROVED** [review]
Approved

**Review by zainm97 — APPROVED** [review]

**Pull request merged**